test/fixtures/rules/std_rule_cases/L064.yml

```diff
@@ -259,3 +259,66 @@ test_pass_date_constructor_strings_are_ignored_2:
       L064:
         force_enable: True
         preferred_quoted_literal_style: double_quotes
+
+test_pass_partially_templated_quoted_literals_simple:
+  pass_str: |
+    SELECT "{{ 'a string' }}"
+  configs:
+    core:
+      dialect: bigquery
+    rules:
+      L064:
+        preferred_quoted_literal_style: double_quotes
+
+test_fail_partially_templated_quoted_literals_simple:
+  fail_str: |
+    SELECT '{{ "a string" }}'
+  configs:
+    core:
+      dialect: bigquery
+    rules:
+      L064:
+        preferred_quoted_literal_style: double_quotes
+
+test_pass_partially_templated_quoted_literals_complex:
+  pass_str: |
+    SELECT "this_is_a_lintable_{{ 'string' }}"
+  configs:
+    core:
+      dialect: bigquery
+    rules:
+      L064:
+        preferred_quoted_literal_style: double_quotes
+
+test_fail_partially_templated_quoted_literals_complex:
+  fail_str: |
+    SELECT 'this_is_a_lintable_{{ "string" }}'
+  configs:
+    core:
+      dialect: bigquery
+    rules:
+      L064:
+        preferred_quoted_literal_style: double_quotes
+
+test_fail_partially_templated_quoted_literals_inside_blocks:
+  fail_str: |
+    SELECT
+        {% if true %}
+        '{{ "another_templated_string" }}'
+        {% endif %}
+  configs:
+    core:
+      dialect: bigquery
+    rules:
+      L064:
+        preferred_quoted_literal_style: double_quotes
+
+test_pass_fully_templated_quoted_literals_are_ignored:
```

Comment:
I suggest some additional test cases. These are not the precise test cases, just to give an idea of the areas to test.

Multiple templated slices in one string:
```
SELECT '{{ "project" }}.{{ "dataset" }}.{{"table"}}'
```

Mix of templated and non-templated quotes, template slice spans segments:
```
SELECT '{{ "project' FROM table1" }}
```

Similarly:
```
{{ "SELECT 'proje" -}}ct' FROM table1
```

Reply:
Yeah, good catch. If onside of the quotes are inside the template. This wont work. Adding some logic for this.